lib/Basic/Default/TaskQueue.inc

```diff
@@ -100,10 +102,24 @@ bool TaskQueue::execute(TaskBeganCallback Began, TaskFinishedCallback Finished,
         T->Env.empty() ? decltype(Envp)(None)
                        : decltype(Envp)(llvm::toStringRefArray(T->Env.data()));
 
+    llvm::SmallVector<char, 64> stdoutPath;
+    llvm::SmallVector<char, 64> stderrPath;
+    if (fs::createTemporaryFile("stdout", "tmp",  stdoutPath)
+        || fs::createTemporaryFile("stderr", "tmp",  stderrPath)) {
+      return true;
+    }
+    StringRef stdoutFile = StringRef(stdoutPath.begin(), stdoutPath.size());
+    StringRef stderrFile = StringRef(stderrPath.begin(), stderrPath.size());
+
+    SmallVector<Optional<StringRef>, 3> redirects;
+    redirects.push_back(llvm::None);
+    redirects.push_back(Optional<StringRef>(stdoutFile));
+    redirects.push_back(Optional<StringRef>(stderrFile));
```

Comment:
No need to do any of this type conversion, or to use a SmallVector instead of a plain array. There's an implicit conversion from SmallString (a subclass of SmallVector) to StringRef, and from StringRef to Optional\<StringRef>. So you should be able to just pass `stdoutPath` and `stderrPath` directly here. (We've also brought None into scope, so you don't need the `llvm::` qualification.)

Reply:
That way is much more pleasant!